ui/pages/settings/security-tab/delete-metametrics-data-button/delete-metametrics-data-button.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { CONSENSYS_PRIVACY_LINK } from '../../../../../shared/lib/ui-utils';
+import ClearMetametricsData from '../../../../components/app/clear-metametrics-data';
+import {
+  Box,
+  ButtonPrimary,
+  Icon,
+  IconName,
+  IconSize,
+  PolymorphicRef,
+  Text,
+} from '../../../../components/component-library';
+import {
+  Display,
+  FlexDirection,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  getMetaMetricsDataDeletionTimestamp,
+  getMetaMetricsDataDeletionStatus,
+  getMetaMetricsId,
+  getShowDataDeletionErrorModal,
+  getShowDeleteMetaMetricsDataModal,
+  getLatestMetricsEventTimestamp,
+} from '../../../../selectors';
+import { openDeleteMetaMetricsDataModal } from '../../../../ducks/app/app';
+import DataDeletionErrorModal from '../../../../components/app/data-deletion-error-modal';
+import { formatDate } from '../../../../helpers/utils/util';
+import { DeleteRegulationStatus } from '../../../../../shared/constants/metametrics';
+
+const DeleteMetaMetricsDataButton = ({
+  wrapperRef,
+}: {
+  wrapperRef?: PolymorphicRef<React.ElementType>;
```

Comment:
We talked about updating the typing here to match the common practice in the codebase when working with refs

Reply:
@danjm I have updated these changes  [here]([73f4f14](https://github.com/MetaMask/metamask-extension/pull/24571/commits/73f4f140163f42c3ebc0462c8f24207e498a83f5))